src/components/functional_module/src/plugin_manager.cc

```diff
@@ -175,14 +175,16 @@ ProcessResult PluginManager::ProcessHMIMessage(
   if (application_manager::ProtocolVersion::kHMI == msg->protocol_version()) {
     std::string msg_method;
     // Request or notification from HMI
-    if (value.isMember("method")) {
+    if (value.isMember("method") && value["method"].isString()) {
       msg_method = value["method"].asCString();
       // Response from HMI
-    } else if (value.isMember("result") && value["result"].isMember("method")) {
+    } else if (value.isMember("result") && value["result"].isMember("method") &&
+               value["result"]["method"].isString()) {
       msg_method = value["result"]["method"].asCString();
       // Error response from HMI
     } else if (value.isMember("error") && value["error"].isMember("data") &&
-               value["error"]["data"].isMember("method")) {
+               value["error"]["data"].isMember("method") &&
+               value["error"]["data"]["method"].isString()) {
       msg_method = value["error"]["data"]["method"].asCString();
     } else {
       LOG4CXX_WARN(logger_,
```

Comment:
@indlu I would suggest to extend text in this warning this way:  "... 'method' field was not found, or was containing an invalid string."